src/openvino.cc

```diff
@@ -1172,7 +1191,8 @@ ModelInstanceState::Infer(
     std::vector<TRITONBACKEND_Response*>* responses,
     const uint32_t response_count)
 {
-  RETURN_IF_OPENVINO_ERROR(infer_request_.infer(), "running inference");
+  RETURN_IF_OPENVINO_ERROR(infer_request_.start_async(), "running inference");
+  infer_request_.wait();
```

Comment:
Why break the synchronous call with asynchronous when we are not doing anything meaningful on this thread apart from waiting for the results?
Additionally, how would error be reported with this setup? Can `request.wait()` throw exception? Should it be enclosed within [RETURN_IF_OPENVINO_ERROR](https://github.com/triton-inference-server/openvino_backend/blob/ac8cddacd2d267d87acef92f41af8460d3609d39/src/openvino_utils.h#L60)?

Reply:
If any error happens, it will be captured in start_async call as an exception. wait is not throwing exceptions.
The reason for using async instead of sync call to the backend is in better performance with high concurrency. Despite async and wait are effectively similar to sync calls, the internal OpenVINO scheduler will work more efficiently with async. In result the throughput results should be higher with multiple instances and clients while the latency with a single instance should be almost the same.